staging/cse/windows/configfunc.ps1

```diff
@@ -342,6 +342,92 @@ function Install-OpenSSH {
     Write-Log "OpenSSH installed and configured successfully"
 }
 
+function Config-CredentialProvider {
+    Param(
+        [Parameter(Mandatory=$true)][string]
+        $KubeDir,
+        [Parameter(Mandatory = $false)][string]
+        $CustomCloudContainerRegistryDNSSuffix
+    )
+
+    $CredentialProviderConfPATH = [Io.path]::Combine("$KubeDir", "credential-provider-config.yaml")
+
+    Write-Log "Configuring kubelet credential provider"
+    $azureConfigFile = [io.path]::Combine("$KubeDir", "azure.json")
+
+    $credentialProviderConfig = @"
+apiVersion: kubelet.config.k8s.io/v1
+kind: CredentialProviderConfig
+providers:
+  - name: acr-credential-provider
+    matchImages:
+      - "*.azurecr.io"
+      - "*.azurecr.cn"
+      - "*.azurecr.de"
+      - "*.azurecr.us"
+    defaultCacheDuration: "10m"
+    apiVersion: credentialprovider.kubelet.k8s.io/v1
+    args:
+      - $azureConfigFile
+"@
+
+    if (![string]::IsNullOrEmpty($CustomCloudContainerRegistryDNSSuffix)) {
+        $credentialProviderConfig = @"
+apiVersion: kubelet.config.k8s.io/v1
+kind: CredentialProviderConfig
+providers:
+  - name: acr-credential-provider
+    matchImages:
+      - "*.azurecr.io"
+      - "*.azurecr.cn"
+      - "*.azurecr.de"
+      - "*.azurecr.us"
+      - "*$CustomCloudContainerRegistryDNSSuffix"
+    defaultCacheDuration: "10m"
+    apiVersion: credentialprovider.kubelet.k8s.io/v1
+    args:
+      - $azureConfigFile
+"@
+    }
+    $credentialProviderConfig | Out-File -encoding ASCII -filepath "$CredentialProviderConfPATH"
+}
+
+function Install-CredentialProvider {
+    Param(
+        [Parameter(Mandatory=$true)][string]
+        $KubeDir,
+        [Parameter(Mandatory = $false)][string]
+        $CustomCloudContainerRegistryDNSSuffix
+    )
+
+    try {
+        $KubeletConfigArgsStr=$global:KubeletConfigArgs -join " "
+        # Out of tree credential provider is turned on as a must after 1.30, and is optinal in 1.29, for cluster < 1.29, it's not enabled.
+        # And only when it's enabled, the credential provider flags are set.
+        if ($KubeletConfigArgsStr -Like "*image-credential-provider-config*" -And $KubeletConfigArgsStr -Like "*image-credential-provider-bin-dir*") {

```

Comment:
Since neither is miss, kubelet will raise an error. I think that we should get the values from $global:KubeletConfigArgs, validate them and use the values directly.
Raise a specific error if the value is invalid.

Reply:
Achieved by checking flag values, and when either flag is not well set or only one var is not empty, we'll raise config error

Reply:
This flag check is also protected by test cases.